net-mail/notmuch: Add version bump 0.23.5

GuillaumeSeren opened the pull request:
Add version bump 0.23.5

The related bug issue is here: https://bugs.gentoo.org/show_bug.cgi?id=605254

monsieurp commented on net-mail/notmuch/notmuch-0.23.5.ebuild:
```diff
@@ -0,0 +1,200 @@
+	bindings python mv README README-python || die
+	mv contrib/notmuch-mutt/README contrib/notmuch-mutt/README-mutt || die
+
+	rm -f Makefile.config # assure that new Makefile.config will be generated
```
Missing `die`.

GuillaumeSeren replied:
Ok changed to 
```
rm -f Makefile.config || die # assure that new Makefile.config will be generated
```

monsieurp commented on net-mail/notmuch/notmuch-0.23.5.ebuild:
```diff
@@ -0,0 +1,200 @@
+
+	if use mutt; then
+		pushd contrib/notmuch-mutt || die
+		emake notmuch-mutt.1
```
notmuch -> `${PN}`. This needs addressing across the rest of the ebuild as well please.

GuillaumeSeren replied:
Not really sure about this one because it is specific to mutt but, I changed it to
```sh
pushd contrib/${PN}-mutt || die
```
So changing from notmuch to ${PN}

monsieurp commented on net-mail/notmuch/notmuch-0.23.5.ebuild:
```diff
@@ -0,0 +1,200 @@
+
+	if use mutt; then
+		pushd contrib/notmuch-mutt || die
+		dobin notmuch-mutt
```
Ditto wrt `${PN}`. Please fix the issue below too.

GuillaumeSeren replied:
ok

GuillaumeSeren commented:
Hey @monsieurp 
thanks for the feedback, it started as a version bump (no changes),
but I will check it later today.

GuillaumeSeren commented:
@monsieurp Ok I've added the changes you suggested.

About changing notmuch to ${PN} I've changed it everywhere so it will e easier to maintain but maybe we should ask the maintainer if he agree ? @aidecoe

aidecoe commented:
Using `PN` instead of a name is not really that useful. If someone really decides on renaming "notmuch" to something else, than e.g. "nmbug" would be renamed as well. On the other hand if I would for some reason decide to maintain 2 flavours of notmuch like I maintain 2 flavours of firejail (firejail and firejail-lts), I would have to rename `PN` to `MY_PN` which does not help at all. So please don't do this renaming.

If you provide some additional changes with version bump, it obscures these fixes. The only fix except of using `PN` everywhere is adding one missing `die` - right? Could you please provide these fixes (and for all ebuilds) in separate PR? (If it's only that single `die` I can fix it myself in no time).

GuillaumeSeren commented:
> On the other hand if I would for some reason decide to maintain 2 flavours of notmuch like I maintain 2 flavours of firejail (firejail and firejail-lts), I would have to rename PN to MY_PN which does not help at all.

@aidecoe I suspected that, so here is the simple version bump.

monsieurp commented:
@aidecoe hi. so `PN` isn't useful at all according to you, if my understanding is correct?

aidecoe commented:
@monsieurp it's not useful in this context. What would be a benefit of renaming every "notmuch" string to "${PN}"? It's more probable that someone will renamed "notmuch-mutt" to e.g. "nmmutt" rather than renaming "notmuch" to something else.

aidecoe commented:
@GuillaumeSeren Thank you. Just include this and it's good to go:

```
-       rm -f Makefile.config # assure that new Makefile.config will be generated
+       # assure that new Makefile.config will be generated
+       rm -f Makefile.config || die
```

GuillaumeSeren commented:
@aidecoe Hey :)
I have added your changes to the ebuild.

aidecoe commented:
Indentation is wrong: 4 spaces instead of 1 tab.

GuillaumeSeren commented:
@aidecoe sorry this is fixed now !

aidecoe commented:
@GuillaumeSeren Please run repoman against your changes every time. Copyright line is out of date.

GuillaumeSeren commented:
@aidecoe ah not sure if I miss something but I try to run repoman manifest or repoman full and I didn't get any changes please explain what command you expect me to run.

aidecoe commented:
`repoman` should show the warning that header is wrong. Just fix it. Please submit when all warnings are fixed.

GuillaumeSeren commented:
@aidecoe Yes but repoman is saying that I am alright
```bash
$ sudo repoman full -d
RepoMan scours the neighborhood...
RepoMan sez: "If everyone were like you, I'd be out of business!"

$ sudo repoman
RepoMan scours the neighborhood...

Note: use --include-dev (-d) to check dependencies for 'dev' profiles

RepoMan sez: "If everyone were like you, I'd be out of business!"

$ sudo repoman manifest
>>> Creating Manifest for /home/gseren/src/free/github/guillaumeseren/gentoo/net-mail/notmuch

$ git status 
On branch net-mail/notmuch-add-bump-version-0.23.5
nothing to commit, working tree clean
```

GuillaumeSeren commented:
Hey @aidecoe 
I have checked the remaining diff due to your last upgrade of the ebuild, and upgraded the date accordingly !

```bash
$ diff -u net-mail/notmuch/notmuch-0.23.4.ebuild net-mail/notmuch/notmuch-0.23.5.ebuild 
--- net-mail/notmuch/notmuch-0.23.4.ebuild      2017-01-17 16:49:58.706148101 +0100
+++ net-mail/notmuch/notmuch-0.23.5.ebuild      2017-01-19 17:10:20.586934261 +0100
@@ -1,4 +1,4 @@
-# Copyright 1999-2017 Gentoo Foundation
+# Copyright 1999-2016 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$

```

> repoman should show the warning that header is wrong. Just fix it. Please submit when all warnings are fixed.

I guess you said this because repoman is *not* actually showing the error, if so I misunderstood that in the first place.

GuillaumeSeren commented:
👍

aidecoe commented:
Done, thank you! (I have completely forgot I could merge PR from command line. I would do that earlier...)

GuillaumeSeren commented:
Great !

monsieurp commented:
@aidecoe Hi! Please stop making merge commits. There are many tools and ways to go about merging contributions and PRs from Github back into Portage. Ask in #gentoo-dev if you need help. Thanks! :)
